streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -688,7 +713,13 @@ private void checkAllPartitions(final Set<String> allSourceTopics,
                 continue;
             }
 
-            standbyTaskIds.addAll(topicGroupTasks);
+            for (final TaskId task : topicGroupTasks) {
+                changelogsByStatefulTask.put(task, topicsInfo.stateChangelogTopics
+                                                  .keySet()
+                                                  .stream()
+                                                  .map(t -> new TopicPartition(t, task.partition))
+                                                  .collect(Collectors.toSet()));
```

Comment:
prop:
```suggestion
                changelogsByStatefulTask.put(
                    task, 
                    topicsInfo.stateChangelogTopics.keySet().stream()
                        .map(topic -> new TopicPartition(topic, task.partition)).collect(Collectors.toSet())
                );

```
The most important change is to rename `t` to `topic` .